v2/api/machinelearningservices/customizations/workspace_extension.go

```diff
@@ -141,3 +145,38 @@ func getContainerRegCreds(keysResp armmachinelearning.ListWorkspaceKeysResult) (
 	}
 	return creds, to.Value(keysResp.ContainerRegistryCredentials.Username)
 }
+
+var _ extensions.ErrorClassifier = &WorkspaceExtension{}
+
+func (ext *WorkspaceExtension) ClassifyError(
+	cloudError *genericarmclient.CloudError,
+	apiVersion string,
+	log logr.Logger,
+	next extensions.ErrorClassifierFunc,
+) (core.CloudErrorDetails, error) {
+	details, err := next(cloudError)
+	if err != nil {
+		return core.CloudErrorDetails{}, err
+	}
+
+	// Override is to treat StorageAccountIsNotProvisioned as retryable for Workspaces
+	// (as we may be waiting for the Storage Account to be provisioned)
+	if isRetryableWorkspaceError(cloudError) {
+		details.Classification = core.ErrorRetryable
+	}
+
+	return details, nil
+}
+
+func isRetryableWorkspaceError(err *genericarmclient.CloudError) bool {
+	if err == nil {
+		return false
+	}
+
+	if err.Code() == "BadRequest" && strings.Contains(err.Message(), "StorageAccountIsNotProvisioned") {
```

Comment:
Do we do same for the KeyVault as well?

Reply:
I couldn't make it fail and couldn't find what the error code might be. The obvious `KeyVaultIsNotProvisioned` didn't turn up any hits.